File: solved/python/Bowling.py
DEFAULT_NUMBER_THROWS = 20

def game(score_list):
    score = 0
    index = 0
    for frame in range(0, 10, 1):
        if (score_list[index] == 10):
            score += 10 + score_list[index+1] + score_list[index+2]
            index-=1
        elif (score_list[index] + score_list[index+1]) == 10:
            score += 10 + score_list[index+2]
        else:
            score += score_list[index] + score_list[index+1]
        index += 2
    return score

File: solved/java/bowling/src/main/java/com/fagerland/Bowling.java
package com.fagerland;

import java.util.List;

public class Bowling {

    public static final int NUM_FRAMES = 10;

    public static int calcScore(List<Integer> scoreList) {
        int score = 0;
        int index = 0;
        for (int i = 0; i < NUM_FRAMES; i++) {
            if (isStrike(scoreList, index)) {
                score += 10 + scoreList.get(index+1) + scoreList.get(index+2);
                index--; // When strike second throw is skipped
            } else if (isSplit(scoreList, index)) {
                score += 10 + scoreList.get(index+2);
            } else {
                score += scoreList.get(index) + scoreList.get(index+1);
            }
            index+=2;
        }
        return score;
    }

    private static boolean isStrike(List<Integer> scoreList, int i) {
        return scoreList.get(i) == 10;
    }

    private static boolean isSplit(List<Integer> scoreList, int i) {
        return (scoreList.get(i) + scoreList.get(i+1)) == 10;
    }
}
